staging/src/k8s.io/apiserver/pkg/storage/value/location_transformer.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package value
+
+import (
+	"bytes"
+	"fmt"
+)
+
+// LocationTransformer holds a transformer interface and the location of the resources it has to be used for.
+type LocationTransformer struct {
+	Location    string
+	Transformer Transformer
+}
+
+type locationTransformers struct {
+	transformers []LocationTransformer
+	err          error
+}
+
+var _ Transformer = &locationTransformers{}
+
+// NewLocationTransformers supports the Transformer interface by first filtering transformers on
+// whether they are configured to write to the given location. This is done by detecting which transformer's
+// location is a prefix of this resource's location, and choosing the most specific of those locations.
+func NewLocationTransformers(err error, transformers ...LocationTransformer) Transformer {
+	if err == nil {
+		err = fmt.Errorf("the provided value not match any of the supported transformers")
+	}
+	return &locationTransformers{
+		transformers: transformers,
+		err:          err,
+	}
+}
+
+// TransformFromStorage collects all possible transformers (based on their location parameter) and
+// attempts to decrypt using each of them
```

Comment:
guessing seems dangerous here. Do we need to add the resource type into the ciphertext spec with the key id?